code.js

```diff
@@ -1,3 +1,22 @@
+// References:
+// https://www.geeksforgeeks.org/binary-search-in-javascript/ for help
+// https://github.com/COSC3020/binary-search-j-halladay5/blob/main/code.js
+
 function binarySearch(list, element) {
+    var start = 0;
+    var end = list.length-1;
+    while(start <= end){
+        var mid = Math.floor((start + end)/2);
+        if(list[mid] === element){
+            while(list[start] != element){
```

Comment:
I see that you're doing this to make the tests pass, but this isn't a great solution. You're essentially doing a linear search from the start when you know that the element you're looking for is close to the mid point.

Reply:
Could you do this more efficiently? You don't need to start at the first element...